### PR TITLE
Add fixed lights and stencil shadows

### PR DESCRIPTION
**Pull request by TFP-dev**
Hi, I have added small function to lower stencil shadows and lights.
mta0
mta5
mta1
mta2
mta3
mta4

### CODE REVIEW

**Comment by Einheit-101** [comment]
This is basically what dr_blendshad resource does, but different :D 
Good improvement.

**Comment by ccw808** [comment]
Do the shadows flicker when:
1) Using a high value in setFarClipDistance?
2) In 16 bit colour mode?

File: Client/game_sa/CSettingsSA.cpp [hunk 1]
```diff
@@ -327,7 +328,22 @@ void CSettingsSA::SetVolumetricShadowsSuspended(bool bSuspended)
...
+}
+
+void CSettingsSA::SetLightsAndShadowsImprovementEnabled(bool bEnable)
+{
+    m_bLightsAndShadowsImprovement = bEnable;
+
+    if (bEnable == true) 
+        MemPut<DWORD>(0x00859934, 1018127695);
+    else
+        MemPut<DWORD>(0x00859934, 1031127695);
```
Comment (Pirulax) [inline Client/game_sa/CSettingsSA.cpp:344]:
```cpp
m_bLightsAndShadowsImprovement = bEnable;
MemPut<float>(0x00859934, bEnable ? 0.0214106123894 : 0.0599999986589);
```
Also I see in IDA this is a float value. Im pretty sure you could just use `MemPut<float>`, or?

Edit: Edited my code to use floats.

**Comment by Woovie** [comment]
setFarClipDistance 100000, 32bit
No perceived flicker
image

I cannot get a 16bit color function to work in Visual Studio, not sure what I need to do for it.

Note that you will see the reflection still looks high up here, I noticed when changing my far clip, it slowly rises. 10000 had barely a perceivable difference. 100000 as you can tell is much more obvious. 1000000 has such a huge offset, the reflection is beyond the curb.

**Comment by patrikjuvonen** [comment]
Thanks for the pull request, we've made further iterations to this and committed at e51c8748eda57308e6a856fb4986a1fd1ac158a3

You were marked as co-author.

**Pull request closed without merging**